core/src/main/resources/common/message/OffsetCommitRequest.json

```diff
@@ -0,0 +1,37 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "name": "OffsetCommitRequest",
+  "validVersions": "0-1",
+  "fields": [
+    {
+      "name": "group",
+      "versions": "0+",
```

Comment:
I suggest to fix the versions to `0-1` for all fields. `OffsetCommitRequest` and `GroupMetadataRequest` are in the same topic and we use the version to differentiate the two so it sounds better to use fix ranges here for the time being. The same for `GroupMetadataRequest`.

A meta comment about this. At the moment, the spec only supports ranges such as `A-B` or `A+`. In this case, I wonder how we will involve this schema as the version `2` is used by the other schema. This is not an issue for now but we may need to extend the spec to support discontinuous ranges in the future.

Reply:
> I suggest to fix the versions to 0-1 for all fields. OffsetCommitRequest and GroupMetadataRequest are in the same topic and we use the version to differentiate the two so it sounds better to use fix ranges here for the time being. The same for GroupMetadataRequest.

I'd like to merge GroupMetadataRequest and OffsetCommitRequest so we can fix the version issue in generated code.

Reply:
I have though about doing this as well and I am not fully convinced by it. What do we gain by doing so? It sounds a bit confusing and error prone as one without the historical context may do mistakes while evolving the two schemas in the future. Moreover, as we still have logic in the code which relies on the version to build up the correct object, we don't gain that much there neither.

As an example, let imagine that we want to add new field `foo` to `GroupMetadata` part of the schema and a new field `bar` to `OffsetCommit` part. We would end up with the following:

```
{
  "type": "data",
  "name": "OffsetCommitKey",
  "validVersions": "0-4",
  "fields": [
    { "name": "group", "type": "string", "versions": "0-4" },
    { "name": "topic", "type": "string", "versions": "0-1, 4" },
    { "name": "partition", "type": "int32", "versions": "0-1, 4" }
    { "name": "foo", "type": "int32", "versions": "3" }
    { "name": "bar", "type": "int32", "versions": "0-1, 4" }
  ]
}
```

v3 would be the new version of the `GroupMetadata` part and v4 the new version of the `OffsetCommit`.

Personally, I find this hard to reason about because it is not easy to see that there are in fact two distinct-overlapping schemas in there which shares the `group` field. I lean towards keeping two separate schemas for the time being. I think that we can revise this when we will have to evolve them.

Reply:
you are right. I will keep two separate schemas.